atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformerJS.kt

```diff
@@ -0,0 +1,310 @@
+                        }
+                    }
+                    var passScope = false
+                    if (field is PropertyGet && field.target.type == Token.THIS) {
+                        passScope = true
+                    }
+                    val args = node.arguments
+                    val inlined = node.inlineAtomicOperation(funcName.toSource(), field, args, passScope)
+                    return !inlined
+                }
+            }
+
+            //remove atomic constructors from classes fields
+            if (node is FunctionCall) {
+                val functionName = node.target.toSource()
+                if (atomicConstructors.contains(functionName)) {
+                    if (node.parent is Assignment) {
+                        val valueNode = node.arguments[0]
+                        (node.parent as InfixExpression).setRight(valueNode)
+                    }
+                    return true
+                }
+            }
+
+            // remove value property call
+            if (node.type == Token.GETPROP) {
+                if ((node as PropertyGet).property.toSource() == "kotlinx\$atomicfu\$value") {
+                    // A.a.value
+                    if (node.target.type == Token.GETPROP) {
+                        val clearField = node.target as PropertyGet
+                        val targetNode = clearField.target
+                        val clearProperety = clearField.property
+                        node.setLeftAndRight(targetNode, clearProperety)
+                    }
+                    // other cases with $receiver.kotlinx$atomicfu$value in inline functions
+                    else if (node.target.toSource() == "\$receiver") {
+                        val rr = ReceiverResolver()
+                        node.enclosingFunction.visit(rr)
+                        if (rr.receiver != null) {
+                            val field = rr.receiver as PropertyGet
+                            node.setLeftAndRight(field.target, field.property)
+                        }
+                    }
+                }
+            }
+            return true
+        }
+    }
+
+    // receiver data flow
+    inner class ReceiverResolver : NodeVisitor {
+        var receiver: AstNode? = null
+        override fun visit(node: AstNode?): Boolean {
+            if (node is VariableInitializer) {
+                if (node.target.toSource() == "\$receiver") {
+                    receiver = node.initializer
+                    return false
+                }
+            }
+            return true
+        }
+    }
+
+    private fun FunctionCall.inlineAtomicOperation(
+        funcName: String,
+        field: AstNode,
+        args: List<AstNode>,
+        passScope: Boolean
+    ): Boolean {
+        var code: String? = null
+        val f = if (passScope) ("scope" + '.' + (field as PropertyGet).property.toSource()) else field.toSource()
+        when (funcName) {
```

Comment:
Use `val code = when { .... else -> null }` here. Btw, using literal strings in the `when` like this is Ok (don't extract them into constants)